pkg/settings/setting.go

```diff
@@ -117,45 +117,104 @@ type NonMaskedSetting interface {
 	ValueOrigin(ctx context.Context, sv *Values) ValueOrigin
 }
 
-// Class describes the scope of a setting in multi-tenant scenarios. While all
-// settings can be used on the system tenant, the classes restrict use on
-// non-system tenants.
-//
-// Settings can only be registered via the Class, e.g.
-// SystemOnly.RegisterIntSetting().
+// Class describes the scope of a setting under cluster
+// virtualization.
 //
 // Guidelines for choosing a class:
 //
-//   - Make sure to read the descriptions below carefully to understand the
-//     differences in semantics.
+//   - Make sure to read the descriptions below carefully to
+//     understand the differences in semantics.
 //
-//   - If the setting controls a user-visible aspect of SQL, it should be a
-//     TenantWritable setting.
+//   - Rules of thumb:
 //
-//   - Control settings relevant to tenant-specific internal implementation
-//     should be TenantReadOnly.
+//     1. if an end-user should be able to modify the setting in
+//     CockroachCloud Serverless, AND different end-users should be
+//     able to use different values, the setting should have class
+//     ApplicationLevel.
 //
-//   - When in doubt, the first choice to consider should be TenantReadOnly.
+//     2. if a setting should only be controlled by SREs in
+//     CockroachCloud Serverless, OR a single value must apply to
+//     multiple tenants (virtual clusters) simultaneously, the setting
+//     must not use class ApplicationLevel.
 //
-//   - SystemOnly should be used with caution: even internal tenant code is
-//     disallowed from using these settings at all.
+//     3. if a non-ApplicationLevel setting is ever accessed by
```

Comment:
Nit: You might want to word this as "if and only if" to avoid splitting the "if" and "only then" part between so many words.  Additionally, you may want to split the last sentence out to its own bullet, as it's a pretty important point that's somewhat buried right now.  Finally instead of "non-ApplicationLevel", can we say "System-level"?

Reply:
> You might want to word this as "if and only if" to avoid splitting the "if" and "only then" part between so many words. 

Done.

> you may want to split the last sentence out to its own bullet

Done.

> Finally instead of "non-ApplicationLevel", can we say "System-level"?

I think we should include both. The 3 points so far only mentioned ApplicationLevel, so I believe the text here should refer to it primarily.